modules/mobile/spec/request/appointments_cancel_request_spec.rb

```diff
@@ -21,210 +21,268 @@
       end
 
-      context 'with valid params' do
-        let(:cancel_id) do
-          Mobile::V0::Appointment.encode_cancel_id(
-            start_date_local: DateTime.parse('2019-11-15T13:00:00'),
-            clinic_id: '437',
-            facility_id: '983',
-            healthcare_service: 'CHY VISUAL FIELD'
-          )
-        end
+      context 'confirmed appointments' do
+        context 'when request body params are missing' do
+          let(:cancel_id) do
+            'abc123'
+          end
 
-        context 'when a valid cancel reason is not returned in the list' do
-          it 'returns bad request with detail in errors' do
-            VCR.use_cassette('appointments/get_cancel_reasons_invalid', match_requests_on: %i[method uri]) do
+          it 'returns a 422 that lists all validation errors' do
+            VCR.use_cassette('appointments/get_cancel_reasons', match_requests_on: %i[method uri]) do
               put "/mobile/v0/appointments/cancel/#{cancel_id}", headers: iam_headers
 
-              expect(response).to have_http_status(:not_found)
-              expect(response.parsed_body['errors'].first['detail']).to eq(
-                'This appointment can not be cancelled online because a prerequisite cancel reason could not be found'
-              )
+              expect(response).to have_http_status(:unprocessable_entity)
+              expect(response.body).to match_json_schema('errors')
+              expect(response.parsed_body['errors'].size).to eq(1)
             end
           end
         end
 
-        context 'when cancel reason returns a 500' do
-          it 'returns bad request with detail in errors' do
-            VCR.use_cassette('appointments/get_cancel_reasons_500', match_requests_on: %i[method uri]) do
-              put "/mobile/v0/appointments/cancel/#{cancel_id}", headers: iam_headers
-              expect(response).to have_http_status(:bad_gateway)
-              expect(response.parsed_body['errors'].first['detail'])
-                .to eq('Received an an invalid response from the upstream server')
-            end
+        context 'with valid params' do
+          let(:cancel_id) do
+            Mobile::V0::Appointment.encode_cancel_id(
+              start_date_local: DateTime.parse('2019-11-15T13:00:00'),
+              clinic_id: '437',
+              facility_id: '983',
+              healthcare_service: 'CHY VISUAL FIELD'
+            )
           end
-        end
-
-        context 'when a appointment cannot be cancelled online' do
-          it 'returns bad request with detail in errors' do
-            VCR.use_cassette('appointments/put_cancel_appointment_409', match_requests_on: %i[method uri]) do
-              VCR.use_cassette('appointments/get_cancel_reasons', match_requests_on: %i[method uri]) do
-                expect_any_instance_of(SentryLogging).not_to receive(:log_exception_to_sentry)
 
+          context 'when a valid cancel reason is not returned in the list' do
```

Comment:
It might be helpful to describe this as "when the cancel reason service does not return a valid reason" or something like that. I hadn't looked at how this works previously and had to look at the code to understand why this was failing.

Reply:
> I think this is good and the change should work as written. As you've written it, I believe this should be testable because you've written it to use the feature flag per user. I think that means you can turn on the flag for judy (who vaos has using v2) and you should go through the v2 flow.
> 
> My one request is that you clean up the tests a little. There are a couple of places where the descriptions don't seem quite right. I think it might be worth trying to remove one of the context layers like "with valid params". IMO having lots of nested contexts tends to make specs harder to read and sometimes results in weird descriptions like: `when appointment can be cancelled when appointment can be cancelled but fails` Which is one of the test contexts here. Talk to me on slack if you want to pair on it.

All good suggestions. Didn't look much at the original specs besides wrapping them in the VAOS v0 context but looking at them more in depth now they definitely needed reworking. Just updated. Let me know what you think.